pyproject.toml

```diff
@@ -0,0 +1,22 @@
+[build-system]
+requires = ["setuptools"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "ml-compiler-opt"
+description = "Tooling for ML in LLVM"
+readme = "README.md"
+requires-python = ">=3.8"
+dependencies = [
```

Comment:
can these be versioned, or shared with the pipenv file?

Reply:
I've added minimum version information to the packages (not specifying exact versions to try and avoid breaking installations with other packages that might have odd version requirements). I would like to share it with the `Pipfile`, but it's not trivial since `setuptools` only supports dynamic dependencies from a `requirements.txt`. I'll look into at some point, but it seems fairly standard to duplicate the deps here (e.g., [here](https://github.com/tensorflow/tensorflow/blob/321e541cbda1b4e473d264b4fe9bdd3dc2dbe488/tensorflow/tools/pip_package/setup.py#L84) in Tensorflow).